lib/spack/spack/version.py

```diff
@@ -593,7 +611,6 @@ def __init__(self, vlist=None):
                 else:
                     self.versions = [vlist]
             else:
-                vlist = list(vlist)
```

Comment:
Why this line has been removed ?

Reply:
Because vlist is only ever iterated by an immediately following for loop which either copies or converts all of its contents. There's no reason to loop over it twice to put it into a temporary list first, or at least none that I could think of or find by testing.